### PR TITLE
[IMPROVEMENT] Add DB migrations support to sample-platform

### LINKED ISSUE #285: [PROPOSAL] Setup Database Migrations
Please prefix your issue with one of the following: [BUG], [PROPOSAL], [QUESTION].

Sample platform commit (found at the bottom of each page) : **X.X.X**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [ ] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [ ] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [ ] I have checked that the issue I'm posting isn't already reported.
- [ ] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/sample-platform/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/sample-platform/issues)
- [ ] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never visited/used the platform.
- [ ] I have used the platform just a couple of times.
- [ ] I have used the platform extensively, but have not contributed previously.
- [ ] I am an active contributor to the platform.

---
From the existing two PRs: #257 and #253 , choose one method to use for database migration and setup for future use along with documentation on how to do migrations.

References:
- https://rollout.io/blog/database-migration/
- https://www.compose.com/articles/schema-migrations-with-alembic-python-and-postgresql/
- https://realpython.com/flask-by-example-part-2-postgres-sqlalchemy-and-alembic/#install-requirements
- https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-iv-database
- https://flask-migrate.readthedocs.io/en/latest/

### PR DESCRIPTION
**Pull request by thealphadollar**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

- We are using flask-migrate and embedding it in our flask run.py itself.
- Before using the commands provided, one needs to set env variable `FLASK_APP=/path/to/run.py`
- It is required since we may be performing schema update in the future and do not want to do it manually like now.
- Now we have a git-like management of database.
- We have to manually migrate the database on a change.

@canihavesomecoffee Actions to perform on the server before merging this PR.

- Install `flask-migrate`
- Set `FLASK_APP=/path/to/run.py` environment variable

Actions to perform once the merge is deployed.

- Set alembic migration head using `flask db stamp head`

Thanks to @T1duS for working on this during GCI18 and providing me with options and code sample in #257 and #253 

closes #257
closes #253 
fixes #285

### CODE REVIEW

**Comment by thealphadollar** [comment]
@canihavesomecoffee We can safely ignore PEP8Speaks issues here since for some reason it is not in sync with pycodestyle rc standard.

I've created an issue at PEP8Speaks. 
https://github.com/OrkoHunter/pep8speaks/issues/133

Do you want me to fix it there?

File: migrations/env.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+from alembic import context
+# add your model's MetaData object here
+# for 'autogenerate' support
+# from myapp import mymodel
+# target_metadata = mymodel.Base.metadata
+from flask import current_app
+from sqlalchemy import engine_from_config, pool
+
+# this is the Alembic Config object, which provides
```
[the remark this answers is not in the document]
Comment (thealphadollar) [reply migrations/env.py]: These files are auto-generated and though I can change them, it's not recommended.

File: migrations/env.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+from __future__ import with_statement
+
+import logging
+from logging.config import fileConfig
+
+from alembic import context
+# add your model's MetaData object here
```
[the remark this answers is not in the document]
Comment (thealphadollar) [reply migrations/env.py]: These files are auto-generated and though I can change them, it's not recommended.

File: migrations/alembic.ini [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# A generic, single database configuration.
+
+[alembic]
+# template used to generate migration files
```
[the remark this answers is not in the document]
Comment (thealphadollar) [reply migrations/alembic.ini]: These files are auto-generated and though I can change them, it's not recommended.

File: migrations/versions/6b335fbd58ab_.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+    sa.ForeignKeyConstraint(['test_id'], ['test.id'], onupdate='CASCADE', ondelete='CASCADE'),
+    sa.PrimaryKeyConstraint('test_id', 'regression_test_id', 'regression_test_output_id'),
+    mysql_engine='InnoDB'
+    )
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_table('test_result_file')
```
[the remark this answers is not in the document]
Comment (thealphadollar) [reply migrations/versions/6b335fbd58ab_.py]: This should NOT be changed as it changes how the code behaves and it is autogenerated and may break alembic.

[the remark this answers is not in the document]
Comment (satyammittal) [reply migrations/versions/6b335fbd58ab_.py]: ok then it is fine.

**Pull request merged**